### PR TITLE
WAZO-2363 Migrate /dhcpd_update using FastAPI

### PR DESCRIPTION
**Pull request by xxbillouxx**
As Python 2 is no more maintained, this endpoint is migrated to Python 3
with FastAPI.

### CODE REVIEW

File: wazo_sysconfd/config.py [hunk 1]
```diff
@@ -58,6 +58,7 @@
     },
     'enabled_plugins': {
         'status': True,
+        'dhcp_update': True,
```
Comment (sduthil) [inline wazo_sysconfd/config.py:61]: Please keep the keys in alphabetical order.

File: wazo_sysconfd/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+from http.server import BaseHTTPRequestHandler
+
+
+class HttpReqError(Exception):
+    def __init__(self, code: int, message: str = None):
+        self.code = code
+        self.message = message or BaseHTTPRequestHandler.responses[code][1]
```
Comment (sduthil) [inline wazo_sysconfd/exceptions.py:11]: The dependency on `BaseHTTPRequestHandler` looks brittle to me: what if `code` is not a standard code? Furthermore, returning a standard message is not very useful.

File: wazo_sysconfd/plugins/dhcp_update/http.py [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+from fastapi import APIRouter
 import subprocess
-from xivo.http_json_server import register, CMD_R, HttpReqError
+
+from wazo_sysconfd.exceptions import HttpReqError
+
+router = APIRouter()
+
 
 DHCPD_UDPATE_COMMAND = ['dhcpd-update', '-dr']
```
Comment (sduthil) [inline wazo_sysconfd/plugins/dhcp_update/http.py:12]: The `http.py` module should only care about the HTTP-side of the request. The actual work should be done in a `service.py` module that is independent of the HTTP protocol.

**Review by sduthil — CHANGES_REQUESTED** [review]

File: wazo_sysconfd/exceptions.py (hunk 2 above)
Comment (xxbillouxx) [reply wazo_sysconfd/exceptions.py:11]: I took this line from the original HttpReError (https://github.com/wazo-platform/xivo-lib-python/blob/d0f4a907aa5d35385f1ae1d70f58f2f794ab91fd/xivo/http_json_server.py#L81)
At least I will not break something.
If we need to improve that, so I think we should create another PR, not linked to the FastAPI migration.
Let me know what you think about that.

**Comment by xxbillouxx** [comment]
recheck

**Comment by xxbillouxx** [comment]
recheck

**Comment by xxbillouxx** [comment]
recheck

File: wazo_sysconfd/plugins/dhcp_update/http.py (hunk 3 above)
Comment (xxbillouxx) [reply wazo_sysconfd/plugins/dhcp_update/http.py]: moved as suggested. thx.

**Review by sduthil — APPROVED** [review]

**Pull request merged**